debezium-core/src/main/java/io/debezium/rest/MetricsResource.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.rest;
+
+import java.util.List;
+
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+
+import org.apache.kafka.connect.connector.Connector;
+import org.json.simple.JSONObject;
+
+import io.debezium.rest.jolokia.JolokiaAttributes;
+import io.debezium.rest.jolokia.JolokiaClient;
+
+public interface MetricsResource {
+    Connector getConnector();
+
+    String getAttributesFilePath();
+
+    List<JSONObject> getMetrics(String connectorName, JolokiaClient client, List<String> attributes);
+
+    String CONNECTOR_METRICS_ENDPOINT = "/connectors/{connector-name}/metrics";
+
+    @GET
+    @Path(CONNECTOR_METRICS_ENDPOINT)
+    @Produces(MediaType.APPLICATION_JSON)
+    default List<JSONObject> getConnectorMetrics(@PathParam("connector-name") String connectorName) {
+        if (getConnector() == null) {
+            throw new RuntimeException("Unable to fetch metrics for connector " + connectorName + " as the connector is not available.");
+        }
+        JolokiaClient jolokiaClient = new JolokiaClient();
```

Comment:
Why do we need to use Jolokia? Can't we just use `MBeanServer` to query the metrics as we are running in the same JVM?

Reply:
Yes we can, but in order to quickly finish the migration from the UI backend to the REST extensions we agreed on a temporary more-or-less 1:1 migration from the UI backend. I filed [DBZ-7177](https://issues.redhat.com/browse/DBZ-7177) for this back then.

Reply:
@rk3rn3r As suggested by @jpechane to use`MBeanServer` directly instead in this code change instead of just adding Jolokia and then removing it in future release. Also some metrics are not available and it is difficult to find the root cause with the current changes. So I am reworking on this to use the `MBeanServer` and remove Jolokia implementation. Hopefully it will help to identify the problem with SQL Server metrics.

Reply:
We agreed on dropping Jolokia for several reasons, for example to keep the additional dependency out of connector modules.